test/jdk/sun/net/www/http/KeepAliveCache/B5045306.java

```diff
@@ -63,150 +66,166 @@
 
 public class B5045306 {
     static HttpServer server;
-
-    public static void main(String[] args) {
-        startHttpServer();
-        clientHttpCalls();
-    }
+    static ExecutorService executor = Executors.newSingleThreadExecutor();
 
     public static void startHttpServer() {
         try {
             server = HttpServer.create(new InetSocketAddress(InetAddress.getLocalHost(), 0), 10, "/", new SimpleHttpTransactionHandler());
-            server.setExecutor(Executors.newSingleThreadExecutor());
-            server.start();
         } catch (IOException e) {
-            e.printStackTrace();
+            throw new RuntimeException(e);
         }
+        server.setExecutor(executor);
+        server.start();
+    }
+
+    public static void stopHttpServer() {
+        server.stop(1);
+        executor.shutdown();
     }
 
-    public static void clientHttpCalls() {
+    public static void clientHttpCalls() throws Exception {
         List<Throwable> uncaught = new ArrayList<>();
         Thread.setDefaultUncaughtExceptionHandler((t, ex) -> {
             uncaught.add(ex);
         });
-        try {
-            System.out.println("http server listen on: " + server.getAddress().getPort());
-            String hostAddr =  InetAddress.getLocalHost().getHostAddress();
-            if (hostAddr.indexOf(':') > -1) hostAddr = "[" + hostAddr + "]";
-            String baseURLStr = "http://" + hostAddr + ":" + server.getAddress().getPort() + "/";
-
-            URL bigDataURL = new URL (baseURLStr + "firstCall");
-            URL smallDataURL = new URL (baseURLStr + "secondCall");
-
-            HttpURLConnection uc = (HttpURLConnection)bigDataURL.openConnection(Proxy.NO_PROXY);
-
-            //Only read 1 byte of response data and close the stream
-            InputStream is = uc.getInputStream();
+        System.out.println("http server listens on: " + server.getAddress().getPort());
```

Comment:
Maybe move this log message to the line after where we call server.start() and perhaps print the `server.getAddress()` instead of just the port.

Reply:
Done